testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java

```diff
@@ -3988,6 +3988,77 @@ void checkPeriodPredicate(Checker checker) {
     sql(sql).ok(expected);
   }
 
+  @Test void testTableFunction() {
+    final String sql = "select * from table(score(table orders))";
+    final String expected = "SELECT *\n"
+        + "FROM TABLE(`SCORE`((TABLE `ORDERS`)))";
```

Comment:
TABLE seems to end up with an extra set of parentheses. something is wrong with how TABLE is unparsed.

Reply:
Good catch.
The first set of parentheses are added when to unparse operands of 'SCORE' function. 
The extra set of parentheses are added when to handle the first operand, which is a SqlCall 'EXPLICT_TABLE' because it's a SqlKind.EXPRESSION.
It makes sense to avoid extra set of parentheses if the function only has one parameter and this parameter is input table.  I've updated.